conda/plugins/manager.py

```diff
@@ -181,6 +182,22 @@ def yield_command_hook_actions(self, hook_type: CommandHookTypes, command: str):
             if command in command_hook.run_for:
                 yield command_hook.action
 
+    def disable_external_plugins(self) -> None:
+        """
+        Disables all currently registered plugins except built-in conda plugins
+        """
+        for registered_name, plugin in self.list_name_plugin():
+            # retrieves the name of the plugin module
+            if isinstance(plugin, types.ModuleType):
+                # the plugin is actually a module, which should have a __name__
+                name: str | None = getattr(plugin, "__name__", None)
+            else:
+                # disable all plugins that can't be easily identified
+                # via their __module__ name (e.g. a class)
+                name: str | None = getattr(plugin, "__module__", None)
```

Comment:
`module.__name__` is _very_ easily spoofed (can just say `__name__ = "foo"` in module) so `module.__spec__.name` is safer

and `object.__module__` is just a copy of `module.__name__` so it suffers from the same spoofing problem so I'm thinking we should consider doing:

```suggestion
            module = inspect.getmodule(plugin)
            if name is None:
                # probably not good to raise an exception here,
                # perhaps better to just block any plugin where
                # we cannot detect a safe name? or perhaps this
                # is the only case where we fallback to the
                # spoofable name?
                raise Exception(...)
            name = module.__spec__.name
```

Reply:
resolved by https://github.com/conda/conda/pull/12869 and https://github.com/conda/conda/pull/12824/commits/a0a188ef5386c98b3334af6d5a4e40f7b246e2ff